### PR TITLE
Match quorum intersection algo with paper

### LINKED ISSUE #7: 🦆Add static type checker to pytest/travis🦆
e.g.. https://github.com/python/mypy or https://github.com/google/pytype
and introduce type hints.

### PR DESCRIPTION
**Pull request by liamsi**
TODOs
- [x] add more tests (copy over from lachowski)
- [x] remove duplicate code (when all tests pass)
- [ ] intactness (could be separate PR though)

(also closes: #7)

### CODE REVIEW

File: stellarobservatory/quorum_intersection.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+def quorum_intersection(fbas: Tuple[Callable[[Set[Type], Type], bool], Set[Type]]):
+    """Takes an FBAS with set of nodes V and returns True iff F has quorum intersection.
+    It prints two disjoint quorums otherwise."""
+    is_slice_contained, all_nodes = fbas
+    len_all_nodes = len(all_nodes)
+    for quorum in traverse_min_quorums(is_slice_contained, set(), all_nodes, len_all_nodes):
+        greatest_q = greatest_quorum(is_slice_contained, all_nodes.difference(quorum), set())
+        if greatest_q != set():
+            logging.info("Found two disjoint quorums: %s, %s", quorum, greatest_q)
+            return False
```
Comment (andrenarchy) [inline stellarobservatory/quorum_intersection.py]: I don't think we should print anything here but instead do `return False, quorum, greatest_q`

Comment (liamsi) [reply stellarobservatory/quorum_intersection.py]: OK 👍

Comment (andrenarchy) [reply stellarobservatory/quorum_intersection.py]: Bonus points: in the tests you could also check whether the returned sets are quorums and don't have intersections :)

Comment (liamsi) [reply stellarobservatory/quorum_intersection.py]: b2dcea3

**Pull request merged**